src/main/java/com/microsoft/aad/msal4j/HttpUtils.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import java.util.List;
+import java.util.Map;
+
+class HttpUtils {
+
+    static String headerValue(Map<String, List<String>> headers, String headerName){
+
+        if(headerName == null || headers == null){
+            return null;
+        }
+
+        List<String> headerValue = headers.get(headerName);
+
+        if(headerValue == null || headerValue.isEmpty()){
+            return null;
+        }
+
+        return headerValue.get(0);
```

Comment:
This seems a little confusing - in the case of an HTTP header where there are multiple values, e.g.: `x-ms-clitelem: 1,0,0,,` this will only return `1`. 

One way to work around this in my adapter is to put `1,0,0,,` in one header - but if so, why not just take a header parameter of type `Map<String, String>` instead?

Reply:
@jianghaolu  Thanks for reviewing, this is a good point. I had originally given this some thought and decided to go with `Map<String, List<String>>` as this is what some http clients seem to expose as the type for headers (for example HttpsURLConnection). 

I missed this when testing as it seems that most headers returned by AAD are returned as strings and not lists (for example when testing with HttpsUrlConnection, OkHttpClient, and Apache's http client `x-ms-clitelem: 1,0,0,,` is  returned as `["1,0,0,,"]` and not `["1", "0","0",""]`). But the point still stands, we would be dropping information for any headers returned as lists so this needs to be updated. 

Seems to me that the most straightforward way is to to return something along the lines of ` String.join(",", headerValue);` We could also change the type to `Map<String, String>` as you suggest, but that would would put the work on the developer to convert multi valued headers to strings, and I'm leaning towards just doing it internally. If there are any other reasons why you think type should be `Map<String, String>` let me know and we can discuss.